react/features/calendar-sync/components/MeetingList.native.js

```diff
@@ -0,0 +1,237 @@
+    }
+
+    _onRefresh: () => void
+
+    /**
+     * Callback to execute when the list is doing a pull-to-refresh.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onRefresh() {
+        const { dispatch } = this.props;
+
+        dispatch(refreshCalendarEntryList());
+    }
+
+    _toDisplayableItem: Object => Object
+
+    /**
+     * Creates a displayable object from an event.
+     *
+     * @private
+     * @param {Object} event - The calendar event.
+     * @returns {Object}
+     */
+    _toDisplayableItem(event) {
+        return {
+            key: `${event.id}-${event.startDate}`,
+            lines: [
+                event.url,
+                this._toDateString(event)
+            ],
+            title: event.title,
+            url: event.url
+        };
+    }
+
+    _toDisplayableList: () => Array<Object>
+
+    /**
+     * Transforms the event list to a displayable list
+     * with sections.
+     *
+     * @private
+     * @returns {Array<Object>}
+     */
+    _toDisplayableList() {
+        const { _eventList, t } = this.props;
+        const now = Date.now();
+        const nowSection = NavigateSectionList.createSection(
+            t('calendarSync.now'),
+            'now'
+        );
+        const nextSection = NavigateSectionList.createSection(
+            t('calendarSync.next'),
+            'next'
+        );
+        const laterSection = NavigateSectionList.createSection(
+            t('calendarSync.later'),
+            'later'
+        );
+
+        if (_eventList && _eventList.length) {
+            for (const event of _eventList) {
+                const displayableEvent = this._toDisplayableItem(event);
+
+                if (event.startDate < now && event.endDate > now) {
+                    nowSection.data.push(displayableEvent);
+                } else if (event.startDate > now) {
+                    if (nextSection.data.length
+                    && nextSection.data[0].startDate !== event.startDate) {
+                        laterSection.data.push(displayableEvent);
+                    } else {
+                        nextSection.data.push(displayableEvent);
+                    }
+                }
+            }
+        }
+
+        const sectionList = [];
+
+        for (const section of [
+            nowSection,
+            nextSection,
+            laterSection
+        ]) {
+            if (section.data.length) {
+                sectionList.push(section);
+            }
+        }
+
+        return sectionList;
+    }
+
+    _toDateString: Object => string
+
+    /**
+     * Generates a date (interval) string for a given event.
+     *
+     * @private
+     * @param {Object} event - The event.
+     * @returns {string}
+     */
+    _toDateString(event) {
+        /* eslint-disable max-len */
+        return `${getLocalizedDateFormatter(event.startDate).format('lll')} - ${getLocalizedDateFormatter(event.endDate).format('LT')}`;
+        /* eslint-enable max-len */
```

Comment:
This would be a case where I recommend intermediary variables to state in english what each part of the date string represents. That would also by the need to disable linting. Although I understand some people do prefer inlining.

Reply:
Agreed, thanks!